src/fitnesse/responders/versions/VersionSelectionResponder.java

```diff
@@ -12,47 +12,47 @@
 import fitnesse.FitNesseContext;
 import fitnesse.authentication.SecureOperation;
 import fitnesse.authentication.SecureReadOperation;
-import fitnesse.authentication.SecureResponder;
 import fitnesse.html.template.HtmlPage;
 import fitnesse.html.template.PageTitle;
 import fitnesse.http.Request;
 import fitnesse.http.Response;
 import fitnesse.http.SimpleResponse;
-import fitnesse.responders.NotFoundResponder;
+import fitnesse.responders.BasicResponder;
 import fitnesse.wiki.*;
 
 import static fitnesse.wiki.PageData.LAST_MODIFYING_USER;
 import static fitnesse.wiki.PageData.PropertyLAST_MODIFIED;
 
 
-public class VersionSelectionResponder implements SecureResponder {
+public class VersionSelectionResponder extends BasicResponder {
 
   @Override
   public Response makeResponse(FitNesseContext context, Request request) throws Exception {
-    SimpleResponse response = new SimpleResponse();
-    String resource = request.getResource();
-    WikiPagePath path = PathParser.parse(resource);
-    WikiPage page = context.getRootPage().getPageCrawler().getPage(path);
-    if (page == null)
-      return new NotFoundResponder().makeResponse(context, request);
+    WikiPage page = getPage(context, request);
 
     PageData pageData = page.getData();
     List<VersionInfo> versions = getVersionsList(page);
 
     HtmlPage html = context.pageFactory.newPage();
-    html.setTitle("Version Selection: " + resource);
-    html.setPageTitle(new PageTitle("Version Selection", PathParser.parse(resource), pageData.getAttribute(PageData.PropertySUITES)));
+    html.setTitle("Version Selection: " + request.getResource());
+    html.setPageTitle(new PageTitle("Version Selection", PathParser.parse(request.getResource()), pageData.getAttribute(PageData.PropertySUITES)));
     html.put("lastModified", makeLastModifiedTag(pageData));
     html.put("versions", versions);
     html.setNavTemplate("viewNav");
     html.put("viewLocation", request.getResource());
     html.setMainTemplate("versionSelection");
 
+    SimpleResponse response = new SimpleResponse();
     response.setContent(html.html(request));
 
     return response;
   }
 
+  @Override
+  protected String contentFrom(FitNesseContext context, Request request, WikiPage requestedPage) {
```

Comment:
Where/when is this method called (as you overridden `BasicResponder. makeResponse()`)

Reply:
I'm working now on another computer. Primarily I let Intellij implement the required with method and left the implementation with `null`. Which caused a null pointer exception because some code expects to receive a `String `object but instead receiving `null`. But on this computer I don't receive a null pointer exception. I also don't find any code invoking the `contentFrom()` method. So I'm going to remove it.